### PR TITLE
fix/157

### PR DESCRIPTION
**Pull request by Prathit-tech**
### **PR Type**
enhancement, documentation

___

### **Description**
- Removed the Research Assistant script.
- Added new scripts for Competitor Researcher, GitHub to Trello sync, and Research Assistant.
- Updated Investment Analysis and News Summary scripts to use OpenAI and HuggingFace models.
- Added setup scripts for Investment Analysis, News Summary, and Research Assistant environments.
- Added Jupyter notebooks for Investment Analysis, News Summary, and Research Assistant.
- Added example .env files and README documentation for all projects.
- Added requirements.txt for all projects.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>ResearchAssistant.py</strong><dd><code>Remove Research Assistant script.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/ResearchAssistant/ResearchAssistant.py

- Removed the entire script for the Research Assistant agent.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-bddb266f90d52f112fc41f559102ccc4bcc1d9ae4a1f375dea27e32a816722d2">+0/-38</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CompetitiorResearcher.py</strong><dd><code>Add Competitor Researcher script with Notion integration.</code></dd></summary>
<hr>
      
examples/competition_researcher/CompetitiorResearcher.py

<li>Added a new script for Competitor Researcher agent.<br> <li> Includes web scraping, data extraction, and Notion integration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-cf991d44096718c569da1a2af30d58f8299ead723c27c2877380e5def2d3aaa9">+62/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>GithubToTrello.py</strong><dd><code>Add GitHub to Trello sync script.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/github_to_trello/GithubToTrello.py

<li>Added a new script for syncing GitHub TODOs with Trello.<br> <li> Includes Flask server to handle webhooks.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-1108a158050ed6f51eac4afac725b857604172f463e3cceac7c1485f5307df7a">+49/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Update Investment Analysis script with OpenAI and dotenv.</code></dd></summary>
<hr>
      
examples/investment_analysis/main.py

<li>Updated to use OpenAI's ChatGPT model.<br> <li> Added dotenv for environment variable management.<br> <li> Improved agent definitions and task execution.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-534888be32afb72ad04b00fda4f380e09d8485f296b1c46f4fe6156e0d09a48b">+24/-22</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Update News Summary script with HuggingFace and dotenv.</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/main.py

<li>Updated to use HuggingFace's model and dotenv for environment <br>variables.<br> <li> Improved agent execution and result handling.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-dcfaf40b538cd1fb16273ece3619403f2548ea2310a490ce98fe801ce0c4fc6b">+31/-20</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Add Research Assistant script with environment setup.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/main.py

<li>Added a new script for Research Assistant agent.<br> <li> Includes environment setup and task execution.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-be26b69828549bc088d0b3b9ba25c00a8ddb44133d413d3bbeb9cae502d6ca35">+48/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup.sh</strong><dd><code>Add setup script for Investment Analysis.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/investment_analysis/setup.sh

<li>Added setup script for Investment Analysis environment.<br> <li> Includes virtual environment creation and package installation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-8a98427011fd1c13a85e8fea98e2166363a8602225bcb3b2bdad4ec735173f9e">+43/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup.sh</strong><dd><code>Add setup script for News Summary.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/setup.sh

<li>Added setup script for News Summary environment.<br> <li> Includes virtual environment creation and package installation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-29f76bf16760810104ecb55568fe1e5b57ab42400968905226d7a73d391c9326">+47/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup.sh</strong><dd><code>Add setup script for Research Assistant.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/setup.sh

<li>Added setup script for Research Assistant environment.<br> <li> Includes virtual environment creation and package installation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-d4593a8ffe396953a2cfcdcb83cc0fad3fda041e5ee923ede07db7c9b11ea992">+47/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation
</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>investment_analyst.ipynb</strong><dd><code>Add Investment Analysis Jupyter notebook.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/investment_analysis/investment_analyst.ipynb

<li>Added Jupyter notebook for Investment Analysis.<br> <li> Includes detailed steps for setting up and running agents.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-3394b2593e27dc11a1da16849d84481efe275d6188f5e8f9e133666928cef39e">+304/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>news_summary_agent.ipynb</strong><dd><code>Add News Summary Jupyter notebook.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/news_summary_agent.ipynb

<li>Added Jupyter notebook for News Summary agent.<br> <li> Includes detailed steps for setting up and running agents.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-a1f9a5aba7bcfdb09a989c95e5e5a329f990da77aa778bdec53b158da988237a">+312/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>research_assistant.ipynb</strong><dd><code>Add Research Assistant Jupyter notebook.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/research_assistant.ipynb

<li>Added Jupyter notebook for Research Assistant.<br> <li> Includes detailed steps for setting up and running agents.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-432245e5e09a49874628d6ac65cd1ed440d0c983386494c9404bcb7214c5b517">+264/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example .env file for Investment Analysis.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/investment_analysis/.env.example

- Added example .env file for Investment Analysis.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-4c6beeda625897bf73e8d1670d5405f855ba017240097ba98b3bd3498ce824a5">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Readme.md</strong><dd><code>Add README for Investment Analysis project.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/investment_analysis/Readme.md

<li>Added README for Investment Analysis project.<br> <li> Includes setup and execution instructions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-dcdd8d98a55c42033108af7758af0198f26cdc8a01863f8955f4a6d871205c41">+72/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example .env file for News Summary.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/.env.example

- Added example .env file for News Summary.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-84d41cc88996873adab4455894809845420e34d601c2ae31e069b0c95f2c29ec">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>readme.md</strong><dd><code>Add README for News Summary project.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/readme.md

<li>Added README for News Summary project.<br> <li> Includes setup and execution instructions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-bd9e5d53b0ae340d4e51cc2e31feaee13bc3594771161ba5fa1250c57c5fd23c">+72/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example .env file for Research Assistant.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/.env.example

- Added example .env file for Research Assistant.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-8baabe76ce8e19ff61b9d03575aa7b0720ced7b19a687db277c785e7b1fd4849">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Readme.md</strong><dd><code>Add README for Research Assistant project.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/Readme.md

<li>Added README for Research Assistant project.<br> <li> Includes setup and execution instructions.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-1391f284fe497322d25a127feed17be7c13160e23356f5141a16173d683bfcfd">+74/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies
</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add requirements.txt for Investment Analysis.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/investment_analysis/requirements.txt

- Added requirements.txt for Investment Analysis project.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-53e7137c55400c4bc49de0210b93b6b2d5b44e3bac3e5fe4453f696900122c91">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add requirements.txt for News Summary.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/news_summary/requirements.txt

- Added requirements.txt for News Summary project.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-9df32296011cb139cf3294387678c3538bf4dd8b97bedccc52f28c074a1d691a">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add requirements.txt for Research Assistant.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/research_assistant/requirements.txt

- Added requirements.txt for Research Assistant project.

</details>
    

  </td>
  <td><a href="https://github.com/ComposioHQ/composio/pull/159/files#diff-8b1f7f224eba31e7030af6a2a46ffe2715cde3f31563a05ffe716e8e30f887d4">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: examples/investment_analysis/Readme.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   ```
+
+3. **Provide Input When Prompted:**
+   Enter the research topic when prompted by the script.
+
+### Running the IPython Notebook
+
+Alternatively, you can also run the provided IPython Notebook file (`investment_research.ipynb`). Open the notebook in Jupyter Notebook or JupyterLab and follow the instructions within the notebook to execute the project.
+
+By following these steps, you'll set up the environment, install the required dependencies, and run the main.py script successfully. You also have the option to run the IPython Notebook for a different interactive experience.
```
Comment (sohamganatra) [inline examples/investment_analysis/Readme.md:72]: the instead of these?

File: examples/investment_analysis/Readme.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+# Investment Research Project
+
+This project involves setting up and running a system of agents to conduct investment research, analyze stocks, and provide investment recommendations. Follow this guide to set up and run the project.
+
```
Comment (sohamganatra) [inline examples/investment_analysis/Readme.md:5]: More information on how it works and why composio comes (1-2 steps would be helpful)

File: examples/investment_analysis/setup.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Create a virtual environment
+python3 -m venv investment_env
+
+# Activate the virtual environment
+if [[ "$OSTYPE" == "linux-gnu"* || "$OSTYPE" == "darwin"* ]]; then
+  # Linux or macOS
```
Comment (sohamganatra) [inline examples/investment_analysis/setup.sh:8]: Would recommend both of you to merge and get a single setup.sh script.

File: examples/research_assistant/setup.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+# Run Composio login command
+echo "Please log in to Composio:"
+composio login
+
+# Add SERPAPI to Composio
+composio add serpapi
+
+echo "Setup complete. To activate the virtual environment, use the appropriate command for your OS:"
+echo "Linux/macOS: source research_assistant_env/bin/activate"
```
Comment (sohamganatra) [inline examples/research_assistant/setup.sh:44]: Why echo all OS stuff

File: examples/investment_analysis/main.py [hunk 5]
```diff
@@ -1,27 +1,26 @@
...
-)
+# Initialize the language model with OpenAI API key and model name
+llm = ChatOpenAI(openai_api_key=os.getenv("OPENAI_API_KEY", "default_key"), model_name="gpt-4o")
 
-# Define tools for the agents
-composiotoolset = ComposioToolSet()
-tools = composiotoolset.get_actions(actions=[Action.SERPAPI_SEARCH])
+# Setup tools using ComposioToolSet
+composio_toolset = ComposioToolSet()
+tools = composio_toolset.get_tools(apps=[App.SERPAPI])
```
Comment (sohamganatra) [inline examples/investment_analysis/main.py:15]: Add comments specifically on this line, so it's more obvious this is how we imported tools using composio.

**Review by sohamganatra — CHANGES_REQUESTED** [review]

**Review by sohamganatra — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**